### PR TITLE
Update delivery guide for delivery test instructions

### LINKED ISSUE #208: Update developer guide for delivery tests instruction

### PR DESCRIPTION
**Pull request by CJianzhi**
fixes #208 

Update user guide parameters to include prefix for Upcoming and AddDeliveryCommand
Update Cost, Quantity, Product, SupplierIndex, DateTime constraint message to fit parameter names
Update developer guide with delivery test instructions

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -607,7 +605,165 @@ testers are expected to do more *exploratory* testing.
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+### Adding a delivery
+
+1. Adding a delivery with existing supplier
+
+    1. Prerequisites: List all suppliers using the `list` command. Ensure there is at least 1 supplier.
```
Comment (LinkesV) [inline docs/DeveloperGuide.md:612]: jz the list command has been changed

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -607,7 +605,165 @@ testers are expected to do more *exploratory* testing.
...
+
+    1. Test case: `add -d on/18-01-2023 15:00 s/1 pro/bread q/500 g c/5.50`<br>
+       Expected: A delivery paired to the first supplier is added. Details of the delivery shown in the status message.
+
+2. Adding a delivery with invalid supplier
+    1. Test case: `add -d on/18-01-2023 15:00 s/0 pro/bread q/500 g c/5.50`<br>
+       Expected: No delivery is added. Error message indicating that SUPPLIER_INDEX should be a positive number greater than 0 and smaller than total number of suppliers shown in the status message.
+
+3. Adding a delivery with invalid units for QUANTITY
+   1. Prerequisites: List all suppliers using the `list` command. Ensure there is at least 1 supplier.
```
Comment (LinkesV) [inline docs/DeveloperGuide.md:622]: the list command has been modified jian zhi

File: docs/UserGuide.md [hunk 3]
```diff
@@ -248,7 +248,11 @@ Format: `add -d on/DELIVERY_DATE_TIME s/SUPPLIER_INDEX pro/PRODUCT q/QUANTITY kg
 
 Parameters:
 
+<<<<<<< HEAD
+- `on/DELIVERY_DATE_TIME`: Must be in dd-MM-yyyy hh:mm format and must not be blank.
+=======
 - `on/DELIVERY_DATE_TIME`: Must be in dd-mm-yyyy hh:mm format and must not be blank.
+>>>>>>> master
```
Comment (LinkesV) [inline docs/UserGuide.md:255]: I think you commited a merge conflict

**Review by LinkesV — CHANGES_REQUESTED** [review]
I think need resolve the merge conflict that was merged and modify all uses of list instances

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -607,7 +605,165 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Listing all deliveries
+    1. Test case: `list -d`<br>
+       Expected: All added deliveries shown. Success message shown in the status message.
+
+### Marking a delivery
+
+1. Marking a delivery as CANCELLED
+
+    1. Prerequisites: List all deliveries using the `list -d` command. Ensure there is at least 1 delivery with either PENDING or DELIVERED status.
```
Comment (ngjianyi) [inline docs/DeveloperGuide.md:643]: Should this be ensure that the first delivery has a PENDING or DELIVERED status?

**Review by ngjianyi — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (CJianzhi) [reply docs/DeveloperGuide.md]: Yup i have included the check for first delivery

File: docs/UserGuide.md (hunk 3 above)
Comment (CJianzhi) [reply docs/UserGuide.md]: Removed it accordingly

File: src/main/java/seedu/address/model/product/Product.java [hunk 5]
```diff
@@ -9,7 +9,7 @@
  */
 public class Product {
 
-    public static final String MESSAGE_CONSTRAINTS = "Product names should be alphanumeric";
+    public static final String MESSAGE_CONSTRAINTS = "PRODUCT should be alphanumeric";
```
Comment (vinc3leong) [inline src/main/java/seedu/address/model/product/Product.java:12]: I liked how you make sure all MESSAGE_CONSTRAINTS are consistent!

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -607,7 +605,165 @@ testers are expected to do more *exploratory* testing.
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+### Adding a delivery
```
Comment (vinc3leong) [inline docs/DeveloperGuide.md:608]: It would be nice to add a title
### Testing Delivery Commands
before this section!

**Review by vinc3leong — COMMENTED** [review]
LGTM!

**Pull request merged**